src/mbgl/programs/fill_extrusion_program.hpp

```diff
@@ -100,12 +99,9 @@ class FillExtrusionProgram : public Program<
                 // We pack a bool (`t`) into the x component indicating whether it is an upper or lower vertex
                 static_cast<int16_t>(floor(nx * factor) * 2 + t),
                 static_cast<int16_t>(ny * factor * 2),
-                static_cast<int16_t>(nz * factor * 2)
-                
-            }},
-            {{
+                static_cast<int16_t>(nz * factor * 2),
                 // The edgedistance attribute is used for wrapping fill_extrusion patterns
-                e
+                static_cast<int16_t>(e)
```

Comment:
Can `e` ever be > 32,767? If it does overflow an `int16_t`, what happens?

Reply:
That's a good point — in theory it could, with a sufficiently large/complex extrusion contained within one tile (so that the perimeter is > 32,767 tile units), and it does not look good (the foreground right wall has overflowed):
![image](https://user-images.githubusercontent.com/3170312/34222985-92a92454-e58b-11e7-8dbb-69537f300ace.png)
https://github.com/mapbox/mapbox-gl-native/pull/10753/commits/7f51cc09797357c31588fd1251beb9d78b0414a8 handles for that case: while it's not perfect (in that it would cause the pattern not to wrap smoothly around the preceding corner), the patterns already don't align at the vertex where the last edge meets the first edge, and this is enough of an edge case that I feel okay with it. (Added the same fix to gl-js in https://github.com/mapbox/mapbox-gl-js/pull/5895/commits/676b8db2ba426ee95cf14a6533d7b418b2cdeaa7).